### DISCUSSION TITLE
[Dropdown] Disable specific options in dropdown programatically

### DISCUSSION THREAD

**Discussion by GrumpyCrouton** [fomantic/Fomantic-UI#1269] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
I apologize if this is covered somewhere in the documentation, but I can't find it.

I have a modal which, when opened, changes a couple of things in the modal itself depending on which button is clicked. One of these things is a multiselect dropdown.

This dropdown is full of documents assigned to a user (The button clicked determines which user the modal is loaded with details for). Some documents are assigned to users via different method, so I want to disable those documents in my dropdown.

Is there any way to pass a list of document ID's to disable the option for in my dropdown? I could just use the `set exactly` behavior, but I'd rather these options be disabled (not selectable) if possible.

The only information I can find about this online is for disabling the entire select box but this is not what I want.

**Comment by lubber-de** [comment]
If you want to disable some entries of the dropdown, the specific item needs to have the `disabled: true` property set
The following example if taken from the remote store feature, but the item json format is identical if you would use the "change values" behavior
https://fomantic-ui.com/modules/dropdown.html#match-search-query-on-server
```json
    {
        "name"  : "Choice 1",
        "value" : "value1",
        "disabled"  : true
       }
```

**Comment by GrumpyCrouton** [comment]
@lubber-de my trouble here is the dropdown is already created and initialized at this point, I just want to disable the already existing options.

Are you saying I could do this with the `change values` behavior by passing it a list of values similar to your code block?

Edit: It seems this behavior removes all of my options, I just want to disable specific ones.

**Comment by lubber-de** [comment]
If you have an identifier for the specific entry node you can just add the `disabled` class to the entry.

**Comment by lubber-de** [comment, marked as answer]
This might work
```javascript
var myDesiredItem = $('.ui.dropdown').dropdown('get item','valueOfTheItem' );
myDesiredItem.addClass('disabled');
```

**Comment by GrumpyCrouton** [comment]
Yeah, I got your idea to work, but I'm looping through a set of options like this:

```
$.each(docs, function( index, value ) {
    $(DROPDOWN_SELECTOR).dropdown('get item', value.id).addClass('disabled');
});
```
